### PR TITLE
first pass at Documenter docs

### PR DESCRIPTION
**Pull request by cormullion**
Fixes https://github.com/JuliaData/CategoricalArrays.jl/issues/30.

### CODE REVIEW

File: docs/src/using.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+ "Middle"
+ "Young" 
+
+```
+
+## Working with categorical arrays
+
+`categorical(A)` - Construct a categorical array with values from `A`
+
+`compress(A)` - Return a copy of categorical array `A` using the smallest reference type
```
Comment (nalimilan) [inline docs/src/using.md]: "smallest possible"?

File: docs/src/overview.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Overview
+
+This package provides a replacement for [DataArrays.jl](https://github.com/JuliaStats/DataArrays.jl)'s `PooledDataArray` type.
+
+It offers better performance by getting rid of type instability thanks to the `Nullable` type, which is used to represent missing data. It is also based on a simpler design by only supporting categorical data, which allows offering more specialized features (like ordering of categories). See the [IndirectArrays.jl](https://github.com/JuliaArrays/IndirectArrays.jl) package for a simpler array type storing data with a small number of values.
```
Comment (nalimilan) [inline docs/src/overview.md]: I would also keep this paragraph and the previous one in README.md so that people can easily know what this is about.

File: README.md [hunk 3]
```diff
@@ -8,283 +8,10 @@ CategoricalArrays.jl
 [![Coveralls](https://coveralls.io/repos/github/JuliaData/CategoricalArrays.jl/badge.svg)](https://coveralls.io/github/JuliaData/CategoricalArrays.jl)
 [![Codecov](https://codecov.io/gh/JuliaData/CategoricalArrays.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/JuliaData/CategoricalArrays.jl)
 
-Tools for working with categorical variables, both with unordered (nominal variables)
```
Comment (nalimilan) [inline README.md]: I would keep this paragraph so that people can easily know what this is about (even if it's duplicated in the manual).

**Comment by cormullion** [comment]
Will do these edits tomorrow probably. Hope it's helped a bit.

File: src/array.jl [hunk 4]
```diff
@@ -580,13 +580,6 @@ Return levels which appear in `A`, in the same order as [`levels`](@ref)
 """
 function unique end
 
-"""
```
Comment (nalimilan) [inline src/array.jl]: Oh, so the docstring was here originally? Then please keep it here, and not duplicate it below. What I was asking for was actually the current state... Why did you move it?

Comment (cormullion) [inline src/array.jl]: Originally I put the `droplevels!` docstrings before the `droplevels!` function, line 719, rather than before `droplevels`, because it didn't get documented (but I copied rather than moved, so that you could see what I'd done). Then in the edit I removed the original.

**Comment by cormullion** [comment]
Ok, this is as far as I can go—there's the `gh-pages` branch business for you still to do.

(Note that the logo is just a placeholder until [#2](https://github.com/JuliaData/Roadmap.jl/issues/2) is closed.)

File: src/array.jl (hunk 4 above)
Comment (nalimilan) [inline src/array.jl]: OK, sorry, I hadn't spotted the difference between `droplevels!` and `droplevels`. That's just a typo: please just add the `!` in `function droplevels end` (the `droplevels` function isn't supposed to exist and has no methods currently).

**Comment by nalimilan** [comment]
Thanks! That's really appreciated.

**Comment by nalimilan** [comment]
The build succeeded, but it looks like we need Documenter to generate an `index.html` page? Browsing to https://juliadata.github.io/CategoricalArrays.jl/latest/ failed, but https://juliadata.github.io/CategoricalArrays.jl/latest/using.html works.

**Comment by cormullion** [comment]
I think eg Documenter.jl's home page has a HTML Redirect page.

**Comment by cormullion** [comment]
Or perhaps we should rename overview.md and change the make.jl file accordingly.

**Comment by nalimilan** [comment]
The redirect is a good idea, but the links should work without it. Looks like we indeed need to rename `overview.md` to `index.md`. Should probably be added to the Documenter manual.

**Comment by nalimilan** [comment]
OK, the rename fixed the link.
